### PR TITLE
Add test to require docs for action plugins.

### PR DESCRIPTION
**Pull request by mattclay**
##### SUMMARY

Add test to require docs for action plugins.

##### ISSUE TYPE

Feature Pull Request

##### COMPONENT NAME

action-plugin-docs sanity test

##### ANSIBLE VERSION

```
ansible 2.8.0.dev0 (action-plugin-docs 6b8545ef44) last updated 2018/11/08 14:10:40 (GMT -700)
  config file = None
  configured module search path = [u'/Users/mclay/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/mclay/code/mattclay/ansible/lib/ansible
  executable location = /Users/mclay/code/mattclay/ansible/bin/ansible
  python version = 2.7.14 (default, Mar 22 2018, 11:39:16) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

### CODE REVIEW

File: docs/docsite/rst/dev_guide/testing/sanity/action-plugin-docs.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Sanity Tests » action-plugin-docs
```
Comment (acozine) [inline docs/docsite/rst/dev_guide/testing/sanity/action-plugin-docs.rst:1]: Needs the `:orphan:` marker, or the docs build will break.

**Review by acozine — COMMENTED** [review]
+1 to adding this test, but mark the new docs file so we don't get sphinx errors.

File: docs/docsite/rst/dev_guide/testing/sanity/action-plugin-docs.rst (hunk 1 above)
Comment (mattclay) [reply docs/docsite/rst/dev_guide/testing/sanity/action-plugin-docs.rst:1]: The tests pass without adding `:orphan:` and most of the sanity docs pages do not have it. We shouldn't add it without verifying it's required. If it is required, we should figure out why the test passes.

Comment (acozine) [reply docs/docsite/rst/dev_guide/testing/sanity/action-plugin-docs.rst:1]: Huh, weird - I wonder why some require it and some don't? Well, I'm okay with merging this as is, then. And I'll explore the question independently.

**Review by gundalow — CHANGES_REQUESTED** [review]
Should `index.rst` be changed to  `:glob:`, currently it's a hardcoded list (which doesn't include this new `rst` file

**Comment by mattclay** [comment]
@gundalow Which index.rst are you referring to?

**Comment by gundalow** [comment]
dev_guide/testing/sanity/index.rst

**Comment by mattclay** [comment]
@gundalow That `index.rst` is generated by the script:

https://github.com/ansible/ansible/blob/devel/docs/bin/testing_formatter.sh

**Comment by gundalow** [comment]
Oh, 
Given we can `:glob:` I wonder if that's overkill. Anyway, that can be done in a future PR

**Review by gundalow — APPROVED** [review]

**Pull request merged**